### ISSUE TITLE
Removing escript causes `mix escript.build` to not rebuild escript

### ISSUE THREAD

**Issue by dideler** [elixir-lang/elixir#5813]
I'm building a CLI application and used `mix escript.build` to produce an escript executable.
At some point the built escript stopped including changes I was making (updated the `--help` output but it still showed the old help). I also tried to force the build but it didn't make a difference.

I decided to remove the executable and rebuild it. But after removing it, it won't rebuild.

Update: Possibly related, `mix test` is also not working (compiles but doesn't run tests). There may be an issue with my mix environment.

### Environment

* Elixir & Erlang versions: Erlang/OTP 19, Elixir 1.4.2
* Operating system: macOS 10.12.3 (Sierra)

```ex
# mix.exs
defmodule MyApp.Mixfile do
  use Mix.Project

  def project do
    [
      app: :my_app,
      # ...
      escript: [main_module: CLI, name: "app_name"],
   ]
  end

  # ...
end
```

### Current behavior

```shell
$ mix escript.build
Compiling 4 files (.ex)

Generated my_app app
Generated escript app_name with MIX_ENV=dev
$ rm app_name
$ mix escript.build
Compiling 4 files (.ex)
```

### Expected behavior

```shell
$ mix escript.build
Compiling 4 files (.ex)

Generated my_app app
Generated escript app_name with MIX_ENV=dev
$ rm app_name
$ mix escript.build
Compiling 4 files (.ex)

Generated my_app app
Generated escript app_name with MIX_ENV=dev
```

**Comment by josevalim** [comment]
I cannot reproduce this. I am using [`ex_doc`](http://github.com/elixir-lang/ex_doc) and building a escript and then deleting successfully builds a new one. Have you tried deleting _build since you mentioned you are running into issues?

**Comment by dideler** [comment]
Deleting _build/ unfortunately didn't make a difference. This is more likely to be a broken environment than a mix bug - sorry for prematurely opening an issue. I'll troubleshoot via other channels. Thanks for trying to reproduce!

**Comment by dideler** [comment]
Found the issue. In between builds I added a `process` function in my CLI module:

```ex
defmodule CLI do
  def main(args) do
    args |> parse_args |> process
  end

  ...

  defp process(:help), do: IO.puts @moduledoc; System.halt(0)

  ...
end
```

If I multi-line the `process` function, or wrap its body in parentheses, then everything works.
[There's a similar issue on SO.](http://stackoverflow.com/questions/31763989/elixir-multiple-expressions-on-same-line-compiler-error-when-using-do-synta)

What I find weird is that the compiler didn't give any warnings or errors.
